### ISSUE TITLE
Add unit tests and increase coverage

### ISSUE THREAD

**Issue by willingc** [nextstrain/augur#476]
**Context**  
Let's give this great project more test coverage. 💯 

**Description**   
If we **divide and conquer**, we should be able to increase test coverage significantly.

Here's a list of Python files which could benefit from tests or additional tests:

- [ ] align.py - groutr
- [ ] ancestral.py - vanguard737
- [ ] clades.py - willingc
- [ ] distance.py - dharrison-salesforce 
- [ ] export.py - 4rund3v
- [ ] export_v1.py
- [ ] export_v2.py
- [ ] filenames.py
- [ ] filter.py - elebow
- [ ] frequencies.py
- [ ] frequency_estimators.py
- [ ] import.py
- [ ] import_beast.py
- [ ] lbi.py
- [ ] mask.py
- [ ] parse.py - dkain
- [ ] reconstruct_sequences.py - zsailor
- [ ] refine.py
- [ ] sequence_traits.py
- [ ] titer_model.py
- [ ] titers.py
- [ ] traits.py - erikguan
- [ ] translate.py -  CodeSammich 
- [ ] tree.py - cgtopher
- [ ] utils.py
- [ ] validate.py - anouarh
- [ ] validate_export.py
- [ ] version.py - St4rz

To claim a file, edit this message and insert your github username after the filename. Please start by claiming 1 or 2 files. You can always return to claim more.

Once your PR is merged, please check off that the file is covered. Celebrate the increase in tests and coverage and that your contribution makes a difference :tada:

cc/ @huddlej

**Comment by groutr** [comment]
I can work on align.py. I've been reviewing that file for most of the day now.

**Comment by elebow** [comment]
I'll take a stab at `filter.py`.

**Comment by huddlej** [comment]
Thank you all for signing up to write tests!

In addition to unit tests for individual functions in each module, it would be amazing to have tests for the command line interfaces of major augur modules, too. For example, the `filter.py` module is also accessible from the command line like so:

```
augur filter --sequences input.fasta --min-length 25000 --output output.fasta
```

Testing this type of interface reveals issues with the entire module's data flow even if all of the module's other unit tests are passing. It looks like some folks are using pytest-mock to test these CLIs and it would be great to keep moving in this direction!

**Comment by willingc** [comment]
Absolutely @huddlej. We should be enabling coverage on CI which will indicate for each PR the increase in test coverage by file and by PR :sunny:

**Comment by jameshadfield** [comment]
In addition to unit tests for each of the python files above (which are great) it would be amazing to have an example of a test of the actual CLI command itself (e.g. see [this comment](https://github.com/nextstrain/augur/pull/481#issuecomment-604688160) & [this comment](https://github.com/nextstrain/augur/pull/481#issuecomment-604737132)). There are a number of invocations of each command within [./tests/builds](https://github.com/nextstrain/augur/tree/master/tests/builds) which could be adapted (with simplifications) into such tests. An example of how to best do this would allow future contributions to extend the coverage of such CLI tests.

**Comment by dkain** [comment]
I can start with `parse.py`

**Comment by anouarh** [comment]
I can work on `validate.py`

**Comment by cgtopher** [comment]
I can start on `tree.py`

**Comment by St4rz** [comment]
I'll start on version.py.

**Comment by vanguard737** [comment]
Quick bookkeeping question:  `augur/filenames.py` is in the above list, and `augur/ancestral.py` is not.  However, when I run tests locally, I observe that filenames.py is 100% covered, while ancestral.py is not:

```
Name                             Stmts   Miss  Cover   Missing
--------------------------------------------------------------
...
augur/ancestral.py                 100     89    11%   44-59, 77-95, 99-113, 118-203
...
augur/filenames.py                   1      0   100%
```

Confirmed that this reproduces in the Travis build output, e.g. [here](https://travis-ci.com/github/nextstrain/augur/jobs/306625752).  Should this issue's worklist be updated accordingly?  (I'd do it myself, but wasn't sure if ancestral.py had been left off of this issue on purpose - e.g. out of scope, or covered in another issue elsewhere, etc.)

**Comment by willingc** [comment]
@vanguard737 The omission of ancestral.py was accidental. I will add it now and assign it to you in the first message. Thanks!

**Comment by dharrison-salesforce** [comment]
Working down the list, I'll start on `distance.py`.

**Comment by Zsailer** [comment]
I'll work on `reconstruct_sequences.py`

**Comment by dharrison-salesforce** [comment]
Is there a good example dataset that can be used with the distance command? Ran through the [tb_tutorial](https://github.com/nextstrain/augur/blob/master/docs/tutorials/tb_tutorial.md) seeing a distance map I can apply to the tree. I can mock up something based on the examples in the docstrings but I'd prefer to see how it works against a real dataset to sanity check myself.

**Comment by erickguan** [comment]
I'll start with `traits.py`.

**Comment by 4rund3v** [comment]
I can work on export.py

**Comment by huddlej** [comment]
@dharrison-salesforce The distance maps described in the docstrings were developed for our seasonal influenza analyses, so the best place to start with real-world data is the [nextstrain/seasonal-flu repo]( https://github.com/nextstrain/seasonal-flu/). Check out the [distances rule in the Snakefile_base](https://github.com/nextstrain/seasonal-flu/blob/bc50a5b9f137c998771b17e4497b5000fb2517e0/Snakefile_base#L703-L724) for an example of how we run the augur distance command.

If you follow the instructions in the repo's README for setting up example data, you should be able to run a simple build quickly on your local machine. Let me know if you have other questions or issues with these data though.

**Comment by lcwheeler** [comment]
I can work on lbi.py

**Comment by CodeSammich** [comment]
I can work on translate.py

**Comment by huddlej** [comment]
Test coverage has improved so much thanks to the efforts here. I'm going to close this long-running issue in favor of smaller issues for specific modules that we'll work on in the future. Thank you all!
